File: lg-provider-api/lg-order-api/src/main/java/com/lg/order/service/CartService.java
package com.lg.order.service;

import com.baomidou.mybatisplus.core.conditions.query.QueryWrapper;
import com.lg.commons.util.wrapper.Wrapper;
import com.lg.order.model.domain.Cart;
import com.lg.order.model.domain.TbOrderItem;
import com.lg.product.model.domain.TbItem;


import java.util.List;

public interface CartService {

    //添加购物车
    public List<Cart> addGoodsToCartList(List<Cart> cartList, Long itemId, Integer num);
    //查询购物车
    public List<Cart> findCartListToRedis(String username);
    //将购物车保存到redis
    public void saveCartListToRedis(String username, List<Cart> cartList);
    //合并购物车
    public List<Cart> mergeCartList(List<Cart> cartList1, List<Cart> cartList2);
    //根据Id查询商品
    public TbItem findByItemid(Long itemId);
    //删除rdis中的商品
    public void deleteCartToRedis(String username, Long[] itemIds);
    //修改商品数量
    public void updateNum(String username, Long itemId, Integer num);
    //滚动商品广告
    public List<TbItem> findItem();
    //移动商品到我的关注
    public void moveCart(String username, Long[] itemIds);
    //查询移动商品
    public List<Cart> findmoveCart(String username);

    List<Cart> getRedisCartByUserAndIds(String name, Long[] ids);
}

File: lg-controller/lg-order-controller/src/main/resources/static/js/service/cartService.js
app.service('cartService', function ($http) {
    //购物车列表
    this.findCartList = function (username) {
        return $http.get('http://localhost:8083/cart/findCartList?username=' + username);
    }

    //添加数量
    this.addGoodsToCartList = function (itemId, num, username) {
        return $http.post('http://localhost:8083/cart/addGoodsToCartList?itemId=' + itemId + "&num=" + num + "&username=" + username);
    }

    //合计数
    this.sum = function (cartList) {
        var totalValue = {totalNum: 0, totalMoney: 0};
        for (var i = 0; i < cartList.length; i++) {
            var cart = cartList[i];
            for (var j = 0; j < cart.orderItemList.length; j++) {
                var orderItem = cart.orderItemList[j];
                totalValue.totalNum += orderItem.num;
                totalValue.totalMoney += orderItem.totalFee;
            }

        }
        return totalValue;
    }

    //查询库存
    this.findKucun = function (itemId) {
        return $http.get('http://localhost:8083/cart/findKucun?itemId=' + itemId);
    }

    //删除商品
    this.deleteCartToRedis = function (ids, username) {
        return $http.get('http://localhost:8083/cart/deleteCartToRedis?ids=' + ids + "&username=" + username);
    }

    //修改商品数量
    this.updateNum = function (id, num, username) {
        return $http.get('http://localhost:8083/cart/updateNum?id=' + id + "&num=" + num + "&username=" + username);
    }

    //滚动商品
    this.findItemList = function () {
        return $http.get('http://localhost:8083/cart/findItem');
    }

    //移动商品到关注
    this.moveCart = function (ids, username) {
        return $http.get('http://localhost:8083/cart/moveCart?ids=' + ids + "&username=" + username);
    };

    //查询关注商品
    this.findmoveCart = function (username) {
        return $http.get('http://localhost:8083/cart/findmoveCart?username=' + username);
    }

});